assignments/prototypes.js

```diff
@@ -32,16 +56,59 @@
   * should inherit destroy() from GameObject through CharacterStats
   * should inherit takeDamage() from CharacterStats
 */
+
+function Humanoid(obj){
+  CharacterStats.call(this, obj);
+  this.team = obj.team;
+  this.weapons = obj.weapons;
+  this.language = obj.language;
+};
  
+Humanoid.prototype = Object.create(CharacterStats.prototype);
+
+Humanoid.prototype.greet = function() {
+  return `${this.name} offers a greeting in ${this.language}.`
+};
+
+
 /*
   * Inheritance chain: GameObject -> CharacterStats -> Humanoid
   * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
   * Instances of CharacterStats should have all of the same properties as GameObject.
 */
 
+// stretch constructors
+
+function Villian(obj){
+  Humanoid.call(this, obj);
+  this.spell = obj.spell;
+};
+
+Villian.prototype = Object.create(Humanoid.prototype);
+
+Villian.prototype.attack = function (human) {
+  human.healthPoints -= 5;
+  if (human.healthPoints > 0) {
+    return `${this.name} attacks ${human.name} with ${this.spell}! \n ${human.takeDamage()} \n Health points: ${human.healthPoints}`;
+  } else if (human.healthPoints <= 0) {
+           return `${human.destroy()}`;
+         }
+};
```

Comment:
You could get away with just `else` instead of `else if` here, but `else if` does make the code very explicit and easy to understand